### PR TITLE
Fixes for OSM deployment

### PR DESCRIPTION
**Pull request by ahmedwaleedmalik**
Signed-off-by: Waleed Malik <ahmedwaleedmalik@gmail.com>

**What this PR does / why we need it**:
Fixes for OSM deployment

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by ahmedwaleedmalik** [comment]
/hold

More changes are required here.

**Comment by floreks** [comment]
I see that from the API side (handler) it's only adding a new flag to the cluster spec.

/approve

File: pkg/resources/environment.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+package resources
+
+import (
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+// sanitizeEnvVar will take the value of a environment variable and sanatises it.
```
Comment (xmudrii) [inline pkg/resources/environment.go:25]: Nit, fix in a follow-up:

```suggestion
// SanitizeEnvVar will take the value of an environment variable and sanitize it.
```

**Review by xmudrii — APPROVED** [review]
/lgtm
/approve

**Pull request merged**